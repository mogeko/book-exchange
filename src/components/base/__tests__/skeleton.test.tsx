import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Skeleton from "@/components/base/skeleton";

describe("Skeleton", () => {
  it("Create a Pulse Line skeleton", () => {
    const { container } = render(
      <Skeleton.Pulse className="test-pulse">
        <Skeleton.Line className="test-line" />
      </Skeleton.Pulse>
    );
    expect(container).toBeInTheDocument();
    expect(container.querySelector("div.animate-pulse")).toHaveClass(
      "test-pulse"
    );
    expect(container.querySelector("div.animate-pulse div")).toHaveClass(
      "test-line"
    );
  });

  it("snapshot a Pulse container", () => {
    const { container } = render(
      <Skeleton.Pulse>
        <></>
      </Skeleton.Pulse>
    );

    expect(container).toMatchSnapshot();
  });

  it("snapshot a Pulse Line skeleton", () => {
    const { container } = render(<Skeleton.Line className="animate-pulse" />);

    expect(container).toMatchSnapshot();
  });

  it("Create a Pulse Square skeleton", () => {
    const { container } = render(<Skeleton.Square className="animate-pulse" />);
    expect(container).toBeInTheDocument();
    expect(container.querySelector("div")).toHaveClass("animate-pulse");
  });

  it("snapshot a Pulse Square skeleton", () => {
    const { container } = render(<Skeleton.Square className="animate-pulse" />);

    expect(container).toMatchSnapshot();
  });

  it("Create a Pulse Circle skeleton", () => {
    const { container } = render(<Skeleton.Circle className="animate-pulse" />);
    expect(container).toBeInTheDocument();
    expect(container.querySelector("div")).toHaveClass("animate-pulse");
  });

  it("snapshot a Pulse Circle skeleton", () => {
    const { container } = render(<Skeleton.Circle className="animate-pulse" />);

    expect(container).toMatchSnapshot();
  });
});
