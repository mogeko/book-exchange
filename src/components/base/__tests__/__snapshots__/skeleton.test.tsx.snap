// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton snapshot a Pulse Circle skeleton 1`] = `
<div>
  <div
    class="animate-pulse bg-slate-700 rounded-full"
  />
</div>
`;

exports[`Skeleton snapshot a Pulse Line skeleton 1`] = `
<div>
  <div
    class="animate-pulse h-2 bg-slate-700 rounded"
  />
</div>
`;

exports[`Skeleton snapshot a Pulse Square skeleton 1`] = `
<div>
  <div
    class="animate-pulse bg-slate-700 rounded"
  />
</div>
`;

exports[`Skeleton snapshot a Pulse container 1`] = `
<div>
  <div
    class="animate-pulse"
  />
</div>
`;
